services/graph/pkg/service/v0/approleassignments.go

```diff
@@ -84,10 +86,38 @@ func (g Graph) CreateAppRoleAssignment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var role string
+	var oldRole string
 	roles := oldRoles.GetAssignments()
 	if len(roles) > 0 {
-		role = roles[0].GetRoleId()
+		oldRole = roles[0].GetRoleId()
+	}
+
+	client, err := g.gatewaySelector.Next()
+	if err != nil {
+		errorcode.NotAllowed.Render(w, r, http.StatusForbidden, err.Error())
+		return
+	}
+
+	if appRoleAssignment.GetAppRoleId() == settingsService.BundleUUIDRoleUserLight {
+		err := shared.DisablePersonalSpace(r.Context(), client, userID)
+		if err != nil {
+			logger.Error().Any("userID", userID).Err(err).Msg("can't disable the personal space")
+			errorcode.RenderError(w, r, err)
+			return
+		}
+	}
+	if appRoleAssignment.GetAppRoleId() != settingsService.BundleUUIDRoleUserLight {
+		user, err := g.identityCache.GetUser(r.Context(), userID)
```

Comment:
Why do we need to fetch the user here? Can't we get it from the context? Like `revactx.ContextGetUser`? This would spare us one API call.